### ISSUE TITLE
Error when running EKS: "EnsembleKalmanProcess has no field obs_mean"

### ISSUE THREAD

**Issue by JunmiHu** [bgroenks96/SimulationBasedInference.jl#8]
## Description
Hi, sorry for bothering you again. Thank you for the help so far!

Running EKS in the linear ode example produces an error "type EnsembleKalmanProcess has no field obs_mean". The error seems to be occuring in the line shown below:

https://github.com/bgroenks96/SimulationBasedInference.jl/blob/7d5efbcdf39d5db0de47c15470eb03aeae0fbbf5/src/ensembles/eks.jl#L73

## To reproduce
I have attached a minimal code below:
```
using SimulationBasedInference
using OrdinaryDiffEq
import Random

const rng = Random.MersenneTwister(1234);
ode_func(u,p,t) = -p[1]*u;
α_true = 0.2
ode_p = [α_true];
tspan = (0.0,10.0);
odeprob = ODEProblem(ode_func, [1.0], tspan, ode_p)

dt = 0.2
tsave = tspan[begin]+dt:dt:tspan[end];
n_obs = length(tsave);
observable = ODEObservable(:y, odeprob, tsave, samplerate=0.01);
forward_prob = SimulatorForwardProblem(odeprob, observable)

ode_solver = Tsit5()
forward_sol = solve(forward_prob, ode_solver);
true_obs = get_observable(forward_sol, :y)
noise_scale = 0.05
noisy_obs = true_obs .+ noise_scale*randn(rng, n_obs);

model_prior = prior(α=Beta(2,2));
noise_scale_prior = prior(σ=Exponential(0.1));

lik = IsotropicGaussianLikelihood(observable, noisy_obs, noise_scale_prior);
inference_prob = SimulatorInferenceProblem(forward_prob, ode_solver, model_prior, lik);

eks_sol = solve(inference_prob, EKS(), ensemble_size=128, rng=rng, verbose=false)
```

## Stack trace
```
ERROR: type EnsembleKalmanProcess has no field obs_mean
Stacktrace:
  [1] getproperty
    @ ./Base.jl:37 [inlined]
  [2] (::SimulationBasedInference.var"#172#174"{…})(y::SubArray{…})
    @ SimulationBasedInference ~/.julia/packages/SimulationBasedInference/t8DhX/src/ensembles/eks.jl:73
  [3] iterate
    @ ./generator.jl:47 [inlined]
  [4] _collect
    @ ./array.jl:854 [inlined]
  [5] collect_similar(cont::ColumnSlices{…}, itr::Base.Generator{…})
    @ Base ./array.jl:763
  [6] map(f::Function, A::ColumnSlices{Matrix{…}, Tuple{…}, SubArray{…}})
    @ Base ./abstractarray.jl:3285
  [7] ensemblestep!(solver::EnsembleSolver{…})
    @ SimulationBasedInference ~/.julia/packages/SimulationBasedInference/t8DhX/src/ensembles/eks.jl:73
  [8] step!(solver::EnsembleSolver{…})
    @ SimulationBasedInference ~/.julia/packages/SimulationBasedInference/t8DhX/src/ensembles/ensemble_solver.jl:178
  [9] solve!(solver::EnsembleSolver{…})
    @ SimulationBasedInference ~/.julia/packages/SimulationBasedInference/t8DhX/src/ensembles/ensemble_solver.jl:201
 [10] solve(::SimulatorInferenceProblem{…}, ::Vararg{…}; kwargs::@Kwargs{…})
    @ CommonSolve ~/.julia/packages/CommonSolve/JfpfI/src/CommonSolve.jl:23
 [11] top-level scope
    @ ~/SBITesting.jl/example/linearode/original_example.jl:30
```

**Comment by bgroenks96** [comment]
Hi, yes this is due to a breaking change in `EnsembleKalmanProcesses` v1.1.6 (which violates semantic versioning... but I digress).

I tried to fix this already by pegging the version to `1.1.5` but I apparently misunderstood how Julia's `compat` versioning works. This will be fixed in the next revision. In the meantime, you can install the package from the main branch instead of the latest version.

**Comment by bgroenks96** [comment]
Fixed by 744a678c2
